### PR TITLE
fix: Content: SEO Tab - Select Page Parent Dropdown is showing the same route multiple times

### LINKED ISSUE #2504: Content: SEO Tab - Select Page Parent Dropdown is showing the same route multiple times
# Bug
Select Page Parent Dropdown is showing the same route multiple times. In this case both routes could be selected, but one of them gave a save error. See Loom: https://www.loom.com/share/ddfe46d1e50d4862b6b74a99245576dc

In another case, 30 minutes after, another set of the same route appearing twice appeared. This time one of the routes could not be selected. See Loom: https://www.loom.com/share/f056c3dccda345ca8a958c51bcf42a0a

Content Item: https://8-da92b5d8e6-hvp4w4.manager.zesty.io/content/6-f4ec9aa998-llpd07/7-b2c1d6898a-t68b3m/meta

@giseleblair @arzljames also have shared that they believe this is not a bug, not a feature

# Solution
Only show each unique route once. Remove the duplicates. Ensure they are: 
- Selectable
- Can be saved with no errors

### PR DESCRIPTION
**Pull request by glespinosa**
This will remove the duplicate of items and when selected and saved it won't have errors

Closes #2504 

Reference image that causing the duplicate
image

[screencast-8-aaeffee09b-7w6v22.manager.dev.zesty.io_8080-2024.02.07-15_34_35.webm](https://github.com/zesty-io/manager-ui/assets/44116036/8239339c-2856-45fd-aefb-f5d483f4b7c6)

### CODE REVIEW

**Review by theofficialnar — APPROVED** [review]

**Comment by shrunyan** [comment]
@glespinosa while this solves the immediate issue in my mind there is a larger question. Why is there an entry in this list which contains a value that does not match the items zuid. Is that expected? @theofficialnar is this something you think warrants further investigation?

File: src/apps/content-editor/src/app/views/ItemEdit/Meta/ItemSettings/settings/ItemParent.js [hunk 1]
```diff
@@ -218,7 +218,8 @@ function parentOptions(currentItemLangID, path, items) {
         itemData?.web?.path && // must have a path
         itemData?.web.path !== "/" && // Exclude homepage
         itemData?.web.path !== path && // Exclude current item
-        itemData?.meta?.langID === currentItemLangID // display only relevant language options
+        itemData?.meta?.langID === currentItemLangID && // display only relevant language options
+        itemZUID === itemData?.meta?.ZUID // must be equal to ensure the itemZUID is correct zuid format
```
[the remark this answers is not in the document]
Comment (glespinosa) [reply src/apps/content-editor/src/app/views/ItemEdit/Meta/ItemSettings/settings/ItemParent.js:222]: When checking the value of itemZuid the values were not in correct format. When the condition equals to false i remove it in the lists cause it cause duplicates and issues when saving. But I dont have any idea yet why there are itemZUID that are incorrect format in the parentOptions list

[the remark this answers is not in the document]
Comment (glespinosa) [reply src/apps/content-editor/src/app/views/ItemEdit/Meta/ItemSettings/settings/ItemParent.js:222]: @agalin920 
If I removed my condition, Interestingly when initially loaded it doesnt have duplicate with 1 label, while on searching it increases to more than 783 that has label 2...

image

And you can see here that there are duplicates on every path that were causing issues...
image

[the remark this answers is not in the document]
Comment (glespinosa) [reply src/apps/content-editor/src/app/views/ItemEdit/Meta/ItemSettings/settings/ItemParent.js]: @shrunyan @agalin920 
The cause of malforming is in the onSearch in setting parents
While the res.data (response) has an array value 
image

The items that we get from 'const items = useSelector((state) => state.content);' 
is an object with a format of
{
 [zuid] : data
}
image

with that inconsistent format , the **parentOptions** function is confuse that's why there are malform zuid like "1","2" cause it's using the key of an array from the original response.data

**Comment by glespinosa** [comment]
> @glespinosa while this solves the immediate issue in my mind there is a larger question. Why is there an entry in this list which contains a value that does not match the items zuid. Is that expected? @theofficialnar is this something you think warrants further investigation?

https://github.com/zesty-io/manager-ui/pull/2541#discussion_r1496782214

**Review by theofficialnar — APPROVED** [review]

**Pull request merged**